server/app/views/store_product.py

```
@@ -0,0 +1,314 @@
+        return repo.create(request)
+
+    @router.post(
+        "/{source_upc}/create-promotional",
+        response_model=StoreProduct,
+        operation_id="createPromotionalProduct",
+    )
+    async def create_promotional_product(
+        self,
+        source_upc: str,
+        request: CreatePromotionalProduct,
+        repo: StoreProductRepository = Depends(store_product_repository),
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)